c_glib/parquet-glib/arrow-file-writer.cpp

```diff
@@ -32,10 +32,129 @@ G_BEGIN_DECLS
  * @short_description: Arrow file writer class
  * @include: parquet-glib/parquet-glib.h
  *
+ * #GParquetWriterProperties is a class for the writer properties.
  * #GParquetArrowFileWriter is a class for writer Apache Arrow data to
  * file as Apache Parquet format.
  */
 
+typedef struct GParquetWriterPropertiesPrivate_ {
+  parquet::WriterProperties::Builder *builder;
+  GArrowCompressionType compression_type;
+} GParquetWriterPropertiesPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(GParquetWriterProperties,
+                           gparquet_writer_properties,
+                           G_TYPE_OBJECT)
+
+#define GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(object) \
+  static_cast<GParquetWriterPropertiesPrivate *>(      \
+    gparquet_writer_properties_get_instance_private(   \
+      GPARQUET_WRITER_PROPERTIES(object)))
+
+enum {
+  PROP_BUILDER = 1
+};
+
+static void
+gparquet_writer_properties_finalize(GObject *object)
+{
+  auto priv = GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(object);
+
+  delete priv->builder;
+
+  G_OBJECT_CLASS(gparquet_writer_properties_parent_class)->finalize(object);
+}
+
+static void
+gparquet_writer_properties_set_property(GObject *object,
+                                        guint prop_id,
+                                        const GValue *value,
+                                        GParamSpec *pspec)
+{
+  auto priv = GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_BUILDER:
+    priv->builder =
+      static_cast<parquet::WriterProperties::Builder *>(g_value_get_pointer(value));
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+gparquet_writer_properties_init(GParquetWriterProperties *object)
+{
```

Comment:
We can always create `parquet::WriterPropertiesBuilder` here.
We don't need to receive it as an argument of `g_object_new()`.